packages/web/src/hooks/useSubscription.ee.js

```diff
@@ -16,23 +33,38 @@ export default function useSubscription() {
 
       return data;
     },
+    refetchInterval: isPolling ? 1000 : false,
   });
+
   const subscription = data?.data;
 
+  const hasSubscription = !!subscription?.status;
+
+  React.useEffect(
+    function pollDataUntilSubscriptionIsCreated() {
+      if (checkoutCompleted && !hasSubscription) {
+        setIsPolling(true);
+      }
+    },
+    [checkoutCompleted, hasSubscription],
+  );
+
+  React.useEffect(
+    function stopPollingWhenSubscriptionIsCreated() {
+      if (checkoutCompleted && hasSubscription) {
+        setIsPolling(false);
+      }
+    },
+    [checkoutCompleted, hasSubscription],
```

Comment:
This two useEffects could be limited to just one. For example: 

`if (checkoutCompleted) {
        setIsPolling(!hasSubscription);
  }`